walkthroughs/howto-k8s-tls-file-based/README.md

```diff
@@ -0,0 +1,227 @@
+
+```
+apiVersion: appmesh.k8s.aws/v1beta2
+kind: VirtualNode
+metadata:
+  name: green
+  namespace: howto-k8s-tls-file-based
+spec:
+  podSelector:
+    matchLabels:
+      app: color
+      version: green
+  listeners:
+    - portMapping:
+        port: 8080
+        protocol: http
+      healthCheck:
+        protocol: http
+        path: '/ping'
+        healthyThreshold: 2
+        unhealthyThreshold: 2
+        timeoutMillis: 2000
+        intervalMillis: 5000
+      tls:
+        mode: STRICT
+        certificate:
+          file:
+            certificateChain: /tls/colorapp-green_cert_chain.pem
+            privateKey: /tls/colorapp-green_key.pem
+  serviceDiscovery:
+    dns:
+      hostname: color-green.howto-k8s-tls-file-based.svc.cluster.local
+```
+
+The `tls` block specifies a filepath to where the Envoy can find the certificates it expects. In order to encrypt the traffic, Envoy needs to have both the certificate chain and the private key.
+
+Virtual Node `blue` certificate was signed by CA 1 and Virtual node `green` certificate was signed by CA 2.
+
+Virtual Node `front` has client side TLS validation enabled. We will change the certificate chain in the Virtual Node `front` to be signed from CA 1 only and then to both CA 1 and CA2 so verify `front` can initially only communicate with Virtual Node `blue` (signed by CA 1) and later communicate with both `blue` (signed by CA 1) and `green` (signed by CA 2), when uses the certificate chain bundle from CA 1 and CA 2.
+
+Kubernetes Secrets are mounted on the Envoy sidecare by the injector in App Mesh Controller for Kubernetes. This is achieved using annotations on your application Pods. Here's a sample YAML for Green Color app deployment.
+
+
+```
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: green
+  namespace: howto-k8s-tls-file-based
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: color
+      version: green
+  template:
+    metadata:
+      annotations:
+        appmesh.k8s.aws/secretMounts: "colorapp-green-tls:/certs/"
+      labels:
+        app: color
+        version: green
+    spec:
+      containers:
+        - name: app
+          image: 669977933099.dkr.ecr.us-west-2.amazonaws.com/howto-k8s-tls-file-based/colorapp
+          ports:
+            - containerPort: 8080
+          env:
+            - name: "COLOR"
+              value: "green"
+```
+
+To mount secret in Envoy sidecar, add the annotation `appmesh.k8s.aws/secretMounts` with "source-secret:destination_path_to_mount_the_secret". 
+
+For example, the annotation `appmesh.k8s.aws/secretMounts: "colorapp-green-tls:/certs/"` will mount contents of secret `colorapp-green-tls` at `/certs/` in the Envoy container. You can verify this by running the following command.
+
+```bash
+GREEN_POD=$(kubectl get pod -l "version=green" -n howto-k8s-tls-file-based --output=jsonpath={.items..metadata.name})
+kubectl exec -it $GREEN_POD -n howto-k8s-tls-file-based -c envoy -- ls /certs/
```

Comment:
I am a bit confused here. In green VirtualNode description the path starts with "/tls" where as it is "/certs" here?

Reply:
These are two different things. ./tls is the directory in walkthrough where cert setup scripts are. /certs in the mount path in Envoy for the secret

Reply:
Ok, then this below sentence is a bit confusing..

"The `tls` block specifies a filepath to where the Envoy can find the certificates it expects. In order to encrypt the traffic, Envoy needs to have both the certificate chain and the private key."

Reply:
So the filepath where Envoy expects to find the certs is actually in the Pod annotation isn't it?

Reply:
That's correct. I had a typo in the YAML. Fixed it